src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTestBase.cs

```diff
@@ -167,11 +169,14 @@ internal Endpoint[] GetEndpointsFromCompilation(Compilation compilation, bool? e
 
         foreach (var endpoint in endpoints)
         {
-            var sourceKeyMetadata = endpoint.Metadata.FirstOrDefault(metadata => metadata.GetType() == sourceKeyType);
+            var sourceKeyMetadata = endpoint.Metadata.OfType<GeneratedCodeAttribute>().SingleOrDefault();
```

Comment:
We emitted the `SourceKey` originally as a proactive measure to allow other possibly generated code (like the validation generator) to do look-ups on generated endpoints by the source location. Since we don't really need this for .NET 8 (we're not shipping other minimal API-related generators), I'm removing this type from the generated code and reusing the `GeneratedCode` attribute to serve as a marker for when the endpoint is statically generated.

Reply:
I asked this in my review below, but I don't fully understand why we are doing this. Why do we need this marker?

Reply:
It helps us determine at runtime if an endpoint was generated at compile-time or runtime. For example, for determining what generator version an endpoint in a library was generated with compared to the app code consuming it.

We typically populate all attributes that are on an endpoint into its metadata so this makes sense to me assuming something like this:

```
app.MapGet("/", [GeneratedCode] (string name) => ...);
```

At runtime, we can get resolve if the endpoint was generated:

```
endpoint.Metadata.OfType<GeneratedCodeAttribute>();
```